src/search.js

```diff
@@ -51,8 +51,11 @@ export default class Search {
   }
 
   search() {
+    // return all docs if the must obj is empty
     const matches = this.filterResults(
-      this.searchFields(this.query.must),
+      Object.keys(this.query.must).length === 0 ?
```

Comment:
The use case this was mainly meant to support was a search with only a filter rather than an empty search. Having match_all in a query that isn't meant to return all documents seems strange

An alternative implementation would just check for the presence of the must object and then return all if it's missing. 
so the current:
```
query: { 
must: {}, 
filter: [
{term: {published: true}}
]
}
```
would become:

```
query: { 
filter: [
{term: {published: true}}
]
}
```